zulipterminal/ui_tools/buttons.py

```diff
@@ -442,7 +443,12 @@ def update_widget(self, caption: str, display_attr: Optional[str] = None) -> Non
         Overrides the existing button widget for custom styling.
         """
         # Set cursor position next to len(caption) to avoid the cursor.
-        icon = urwid.SelectableIcon(caption, cursor_position=len(caption) + 1)
+        icon = urwid.Pile(
+            widget_list=[
+                urwid.SelectableIcon(caption, cursor_position=len(caption) + 1),
+                urwidgets.Hyperlink(uri=self.link),
+            ]
+        )
```

Comment:
Having the rendering of the link defined in the link button like this would be cleaner, and it may be reasonable to refactor the layout into the button as a prep refactor for this.

However, right now this overall change affects the rendering quite a lot: links are not styled in the same way as the background of the button, or when selected. That makes it difficult to visually identify that the link and number/title are related to the link when many links are present.

I'm not sure if that's a limitation of the Hyperlink?

Reply:
Hello!

> However, right now this overall change affects the rendering quite a lot: links are not styled in the same way as the background of the button, or when selected. That makes it difficult to visually identify that the link and number/title are related to the link when many links are present.
>
> I'm not sure if that's a limitation of the Hyperlink?

Yes, it was.

I just released [v0.2.0](https://github.com/AnonymouX47/urwidgets/releases/tag/v0.2.0) which allows the use of `Attrmap`s (and other means) over `Hyperlink` widgets (and the rendered canvases). See https://github.com/AnonymouX47/urwidgets/pull/3.

Reply:
Replace with:

```py
        hyperlink = urwidgets.Hyperlink(uri=self.link)
        icon = urwid.Pile(
            widget_list=[
                urwid.SelectableIcon(caption, cursor_position=len(caption) + 1),
                # using `hyperlink.pack()[0]` instead of `len(self.link)` in case of wide characters
                urwidgets.TextEmbed((hyperlink.pack()[0], hyperlink)),
            ]
        )
```
to wrap the URL across lines instead of clipping with ellipsis.

The `OSC 8` feature will still work perfectly on any terminal emulator with a spec-compliant implementation and behaviour will remain the same as before this PR on terminal emulators that don't support the feature.